docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -393,13 +403,109 @@ and our ``ConstraintApplicationResult`` can set ``remainingFilter`` to
+                    .collect(toImmutableList());
+
+            // Get information about how to push down every column based on it's
+            // JDBC data type
+            List<ColumnMapping> columnMappings = jdbcClient.toColumnMappings(
+                    session,
+                    columnHandles.stream()
+                            .map(JdbcColumnHandle::getJdbcTypeHandle)
+                            .collect(toImmutableList()));
+
+            // Calculate the domains which can be safely pushed down (supported)
+            // and those which need to be filtered in Trino (unsupported)
+            Map<ColumnHandle, Domain> supported = new HashMap<>();
+            Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+            for (int i = 0; i < columnHandles.size(); i++) {
+                JdbcColumnHandle column = columnHandles.get(i);
+                DomainPushdownResult pushdownResult =
+                    columnMappings.get(i).getPredicatePushdownController().apply(
+                        session,
+                        domains.get(column));
+                supported.put(column, pushdownResult.getPushedDown());
+                unsupported.put(column, pushdownResult.getRemainingFilter());
+            }
+
+            newDomain = TupleDomain.withColumnDomains(supported);
+            remainingFilter = TupleDomain.withColumnDomains(unsupported);
+        }
+
+        // Return empty Optional if nothing changed in filtering
+        if (oldDomain.equals(newDomain)) {
+            return Optional.empty();
+        }
+
+        handle = new JdbcTableHandle(
+                handle.getRelationHandle(),
+                newDomain,
+                ...);
+
+        return Optional.of(
+                new ConstraintApplicationResult<>(
+                    handle,
+                    remainingFilter));
+    }
+
+In this example we are implementing a base class for many JDBC connectors,
+so we have to handle all the nuances of different JDBC-capable databases.
+We need to ensure that if a constraint gets pushed down, it should work
+exactly the same in the underlying data source and produce the same results
+as it would in Trino.
+
+For example, string comparison in some databases is case-insensitive, which
+means for those databases we can't push it down since in Trino string
+comparison is case-sensitive.
+
+To handle push down for all the different JDBC types we need a
+``PredicatePushdownController`` which decides if a column domain
+can be pushed down. In the example code above we obtain it from a
+``JdbcClient`` implementation specific to that database. Note that
+this interface is only needed because we are trying to be generic. In non-JDBC
+databases the rules for pushing down different types would usually be
+implemented directly, without going through a ``PredicatePushdownController``
+interface.
+
+Now let's add expression pushdown to our code, enabled by a session flag:
```

Comment:
Do not use "we" - too informal.
Do not use "usually" (or "could"  or "would") as per google dev style guide. In reference material, writing must be definite.
Do not use directions such as above, below. User following/preceding (g dev style guide)

Also, I have used "interface" below, because you seem to call it an interface in the paragraph. Please ensure that it is being named *precisely* what it is (interface/class/method, etc).

The ``PredicatePushdownController`` interface determines if a column domain can be pushed down in JDBC-compliant data sources. In the preceding example, it is called from a ``JdbcClient`` implementation specific to that database. In non-JDBC-compliant data sources, type-based push downs are implemented directly, without going through the ``PredicatePushdownController`` interface.

Now let's add expression pushdown to our code, enabled by a session flag: